projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts

```diff
@@ -3474,6 +3443,89 @@ describe('IgxGrid - Filtering Row UI actions #grid', () => {
+            input.click();
+            tick(100);
+            fix.detectChanges();
+
+            // Click the today date.
+            const outlet = document.getElementsByClassName('igx-grid__outlet')[0];
+            let calendar = outlet.getElementsByClassName('igx-calendar')[0];
+            const todayDayItem = calendar.querySelector('.igx-calendar__date--current');
+            clickHtmlElemAndBlur(todayDayItem, inputDebugElement);
+            tick(100);
+            fix.detectChanges();
+
+            // Select our newly added chip
+            const chip = GridFunctions.getFilterConditionChip(fix, 0);
+            const chipDiv = chip.querySelector('.igx-chip__item');
+
+            expect(chipDiv.classList.contains('igx-chip__item--selected')).toBeTruthy('initial chip is committed');
+
+            // Focus out
+            clickHtmlElemAndBlur(chip, inputDebugElement);
+            tick(200);
+            fix.detectChanges();
+
+            expect(chipDiv.classList.contains('igx-chip__item--selected')).toBeFalsy('initial chip is not committed');
+            expect(input.value).toBeFalsy('initial input value is present and not committed');
+
+            chip.click();
+            tick(200);
+            fix.detectChanges();
+
+            // Open calendar
+            input.click();
+            tick(100);
+            fix.detectChanges();
+
+            calendar = outlet.getElementsByClassName('igx-calendar')[0];
+
+            // View years
+            const yearView: HTMLElement = calendar.querySelectorAll('.igx-calendar-picker__date')[1] as HTMLElement;
+            yearView.click();
+            tick(100);
+            fix.detectChanges();
+
+            // Select the first year
+            const firstYear: HTMLElement = calendar.querySelectorAll('.igx-calendar__year')[0] as HTMLElement;
+            firstYear.click();
+            tick(100);
+            fix.detectChanges();
+
+            // Select the first day
+            const firstDayItem: HTMLElement = calendar.querySelector('.igx-calendar__date');
+            clickHtmlElemAndBlur(firstDayItem, inputDebugElement);
+            tick(100);
+            fix.detectChanges();
+
+            expect(chipDiv.classList.contains('igx-chip__item--selected')).toBeTruthy('chip is committed');
+
+            // Focus out
+            clickHtmlElemAndBlur(chip, inputDebugElement);
+            tick(200);
+            fix.detectChanges();
+            expect(chipDiv.classList.contains('igx-chip__item--selected')).toBe(false, 'chip is selected');
+
+            // Check if we still have only one committed chip
+            const activeFilterChip: IgxChipComponent = filteringRow.query(By.directive(IgxChipComponent)).componentInstance;
+            const chipsLength = GridFunctions.getAllFilterConditionChips(fix).length;
+
+            expect(chipsLength).toBe(1, 'there is more than one chip');
+            expect(activeFilterChip.selected).toBe(false, 'chip is not committed');
```

Comment:
You can use the same approach for verifying whether the chip is selected as above (verifying the chipDiv class).